tests/test_operations.py

```diff
@@ -0,0 +1,342 @@
+):
+    """Validate ConfigureOperation.format"""
+    exec = configure_operation.format(generation_context)
+    assert isinstance(exec, Command)
+    assert str(mock_src) in exec.command
+    assert configure_cmd in exec.command
+    assert create_final_dest(mock_src, mock_dest) in exec.command
+
+
+def test_init_file_sys_operation_set(file_system_operation_set: FileSysOperationSet):
+    """Test initialize FileSystemOperationSet"""
+    assert isinstance(file_system_operation_set.operations, list)
+    assert len(file_system_operation_set.operations) == 3
+
+
+def test_add_copy_operation(
+    file_system_operation_set: FileSysOperationSet, copy_operation: CopyOperation
+):
+    """Test FileSystemOperationSet.add_copy"""
+    assert len(file_system_operation_set.copy_operations) == 1
+    file_system_operation_set.add_copy(src=pathlib.Path("src"))
+    assert len(file_system_operation_set.copy_operations) == 2
+
+
+def test_add_symlink_operation(
+    file_system_operation_set: FileSysOperationSet, symlink_operation: SymlinkOperation
+):
+    """Test FileSystemOperationSet.add_symlink"""
+    assert len(file_system_operation_set.symlink_operations) == 1
+    file_system_operation_set.add_symlink(src=pathlib.Path("src"))
+    assert len(file_system_operation_set.symlink_operations) == 2
+
+
+def test_add_configure_operation(
+    file_system_operation_set: FileSysOperationSet,
+    configure_operation: ConfigureOperation,
+):
+    """Test FileSystemOperationSet.add_configuration"""
+    assert len(file_system_operation_set.configure_operations) == 1
+    file_system_operation_set.add_configuration(
+        src=pathlib.Path("src"), file_parameters={"FOO": "BAR"}
+    )
+    assert len(file_system_operation_set.configure_operations) == 2
+
+
+# might change this to files that can be configured
+@pytest.fixture
+def files(fileutils):
+    path_to_files = fileutils.get_test_conf_path(
+        osp.join("generator_files", "easy", "correct/")
+    )
+    list_of_files_strs = sorted(glob(path_to_files + "/*"))
+    yield [pathlib.Path(str_path) for str_path in list_of_files_strs]
+
+
+@pytest.fixture
+def directory(fileutils):
+    directory = fileutils.get_test_conf_path(
+        osp.join("generator_files", "easy", "correct/")
+    )
+    yield [pathlib.Path(directory)]
+
+
+@pytest.fixture
+def source(request, files, directory):
+    if request.param == "files":
+        return files
+    elif request.param == "directory":
+        return directory
+
+
+@pytest.mark.parametrize(
+    "dest,error",
+    (
+        pytest.param(123, TypeError, id="dest as integer"),
+        pytest.param("", TypeError, id="dest as empty str"),
+        pytest.param("/absolute/path", TypeError, id="dest as absolute str"),
+        pytest.param(
+            pathlib.Path("relative/path"), ValueError, id="dest as relative Path"
+        ),
+        pytest.param(
+            pathlib.Path("/path with spaces"), ValueError, id="dest as Path with spaces"
+        ),
```

Comment:
Similar "do we want to guard against this?" question here. There is nothing in *nix that is stopping us from doing this